tests/test_indexes.py

```diff
@@ -121,11 +121,20 @@ def test_str_match() -> None:
 
 
 def test_index_rename() -> None:
+    """Test that index rename returns an element of type Index."""
     ind = pd.Index([1, 2, 3], name="foo")
     ind2 = ind.rename("goo")
     check(assert_type(ind2, "pd.Index[int]"), pd.Index, np.integer)
 
 
+def test_index_rename_inplace() -> None:
+    """Test that index rename in-place does not return anything (None)."""
+    ind = pd.Index([1, 2, 3], name="foo")
+    ind2 = ind.rename("goo", inplace=True)
+    check(assert_type(ind2, None), type(None))
```

Comment:
Not totally sure about the structure when there is a None return, will double check with another example from the code.